packages/ai/core/generate-image/generate-image.ts

```diff
@@ -110,20 +110,27 @@ Only applicable for HTTP-based providers.
 }): Promise<GenerateImageResult> {
   const { retry } = prepareRetries({ maxRetries: maxRetriesArg });
 
+  // extract maxImagesPerCall from providerOptions as it's not meant to be passed to `doGenerate()`
+  const [
+    [providerName, { maxImagesPerCall, ...generateProviderOptions } = {}] = [],
+  ] = Object.entries(providerOptions ?? {});
+
   // default to 1 if the model has not specified limits on
   // how many images can be generated in a single call
-  const maxImagesPerCall = model.maxImagesPerCall ?? 1;
+  const maxImagesPerCallWithDefault =
+    (maxImagesPerCall as number) ?? model.maxImagesPerCall ?? 1;
```

Comment:
This is probably the main change. Before we had access to `model.settings` in the `maxImagesPerCall` getter. Now that `settings` has been removed from the `ImageModelV2` constructor, that is no longer the case.